### ISSUE TITLE
Dark mode for Linux version

### ISSUE THREAD

**Issue by mrjohnc** [transmission/transmission#1213]
Pretty please with sugar on top create an option to have a dark theme for the Linux version

**Comment by mikedld** [comment]
You'll have to be more specific, at least specifying which client you're using and maybe proving some screenshots. In general, I'd say it should just work as both GTK and Qt clients should be using system colors coming from your theme. In some edge cases it doesn't but that would still most probably be related to your system configuration; e.g. sometimes one would need to do some additional configuration when using GTK client in Qt-based DE and vice versa, or when not using any DE at all.

**Comment by axelmeunier** [comment]
Using the gtk version on Ubuntu here. Transmission does follow the system theme indeed. What @mrjohnc might have meant is that some gtk apps provide an option to turn on dark or light theme independently of the system theme. [Apostrophe](https://somas.pages.gitlab.gnome.org/apostrophe/) is probably the best implementation of that feature. Other apps such as Celluloid or Fragments provide it as well in the preference settings. It would be nice, although certainly with an "enhancement" tag, if transmission offered that possibility.

**Comment by mrjohnc** [comment]
@mikedld thanks very much for the response. So I'm using it on ChromeOS which doesn't have themes, just plain white for everything... I am able to select a dark theme in v1.0 of Inkscape so I assumed something similar might be possible with Transmission also which I think is what @axelmeunier is referring to.

Thanks again

**Comment by mikedld** [comment]
It's my understanding that programs which are related to text/photo/video editing or viewing historically provided some sort of distraction-free mode, which may or may not have included a dark theme. It's indeed beneficial and easy on your eyes when you have to work in a program for an extended period of time with varying lighing conditions and your OS doesn't support any kind of global light/dark mode switch (which only became a thing quite recently on Windows and macOS; although I'm not mentioning WindowBlinds etc. here). On Linux though, it was always possible to pick a theme that you're comfortable with or create your own, and I find it odd that some people would want some programs to be light and some other programs to be dark... The reason those other programs provide you with dark mode is also because usually they don't have that many native UI elements and it's less of a concern for them to look alien on your desktop.

In any case, any Qt program (Transmission Qt client included) accepts a number of standard command line parameters like `-style` and `-stylesheet` which allow you to change its appearance and disregard global system settings; there's also [qt5ct](https://sourceforge.net/projects/qt5ct) project that provides a style plugin you could inject in a particular program (or globally) using `QT_QPA_PLATFORMTHEME` environment variable. For GTK, similar functionality seems to be available via `GTK_THEME` environment variable.

In conclusion, I'd say we may consider implementing this if GNOME or some other guidelines will start recommending programs to include this kind of option in their preferences. Otherwise it feels wrong to me personally (@ckerr feel free to jump in here).

On a side note, if you need a dark mode in ChromeOS and it's impossible to get one (I have no idea), either consider reporting this to Google or maybe think about using another, less limiting Linux distribution.

**Comment by mrjohnc** [comment]
@mikedld Thank you very much for your thoughtful and thorough reply.

I'm sorry if I'm misunderstood, are you saying I could launch Transmission with an additional command to turn it to dark mode? If so would it be possible to include this command in the readme?

Yes I'd love to use Fedora/Manjaro/whatever works on my laptop but currently there are no Linux drivers for it so I'm stuck with ChromeOS for the time being....

**Comment by ckerr** [comment]
This doesn't seem very useful? Seems like most people who want a dark mode version of Transmission will already have their entire desktop in dark mode, so why bother adding code to let the app have a different mode from the desktop?

Unless one of the other maintainers disagrees, I'm leaning towards closing this as 'declined'
